### PR TITLE
feat: Create bddtest for basic flow

### LINKED ISSUE #260: Create bddtest for basic flow
Basic flow:https://wiki.hyperledger.org/display/ARIES/DID+Exchange+Flow+-+Framework+Go

- Create feature file.
1- Agent A starts
2- Agent B starts
3- Agent A create invitation
4- Agent B receive  invitation
5- Agent A and B Query connection
6- Verify connection record (state should be completed)

### PR DESCRIPTION
**Pull request by fqutishat**
- Add bddtest for did exchange basic flow

closes #260

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -52,6 +53,7 @@ github.com/pmezard/go-difflib v1.0.0/go.mod h1:iKH77koFhYxTK1pcRnkKkqfTogsbg7gZN
 github.com/pquerna/cachecontrol v0.0.0-20180517163645-1555304b9b35 h1:J9b7z+QKAmPf4YLrFg6oQUotqHQeUNWwkvo7jZp1GLU=
 github.com/pquerna/cachecontrol v0.0.0-20180517163645-1555304b9b35/go.mod h1:prYjPmNq4d1NPVmpShWobRqXY3q7Vp+80DqgxxUrUIA=
 github.com/russross/blackfriday v1.5.2/go.mod h1:JO/DiYxRf+HjHt06OyowR9PTA263kcR/rfWxYHBV53g=
+github.com/sirupsen/logrus v1.4.2 h1:SPIRibHv4MatM3XXNO2BJeFLZwZ2LvZgfQ5+UNI2im4=
```
Comment (troyronda) [inline go.sum]: I think we should consider having non-framework directories (like bdd, cmd) directory specify a module to avoid leaking dependencies.

Comment (troyronda) [reply go.sum]: I see that you did. Now I am confused how logrus leaked.

Comment (fqutishat) [reply go.sum]: bdd has it's own module I don't know why go.sum in the root get effected I will check it

Comment (fqutishat) [reply go.sum]: I will check it not suppose to leak

Comment (fqutishat) [reply go.sum]: it's gone now should okay now

File: .gitignore [hunk 2]
```diff
@@ -30,3 +30,5 @@ coverage.txt
 
 # Exclude build directory
 build
+
+test/bdd/db
```
Comment (troyronda) [inline .gitignore]: you might also need to have codecov ignore test directory.

Comment (fqutishat) [reply .gitignore]: ok

File: .codecov.yaml [hunk 3]
```diff
@@ -10,3 +10,6 @@ coverage:
     patch:
       default:
         target: 85%
+
+ignore:
+  - "bdd"  # ignore bdd tests
```
Comment (troyronda) [inline .codecov.yaml]: don't you mean test/bdd?

Comment (fqutishat) [reply .codecov.yaml]: yes

File: test/bdd/agent_steps.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	}
+	// create new did exchange client
+	didexchangeClient, err := didexchange.New(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create new didexchange client: %w", err)
+	}
+
+	a.bddContext.DIDExchangeClients[agentID] = didexchangeClient
+	// wait until inbound http server start
+	time.Sleep(2 * time.Second)
```
Comment (soluchok) [inline test/bdd/agent_steps.go]: please use `listenFor(host string, d time.Duration)` function instead of sleeping

Comment (fqutishat) [reply test/bdd/agent_steps.go]: can you put the logic of this method ```listenFor(host string, d time.Duration)```

Comment (soluchok) [reply test/bdd/agent_steps.go]:
```

func listenFor(host string, d time.Duration) error {
	timeout := time.After(d)
	for {
		select {
		case <-timeout:
			return errors.New("timeout: server is not available")
		default:
			conn, err := net.Dial("tcp", host)
			if err != nil {
				continue
			}
			if err := conn.Close(); err != nil {
				return err
			}
			return nil
		}
	}
}

```

File: test/bdd/features/didexchange_e2e.feature [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#
+# Copyright SecureKey Technologies Inc. All Rights Reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Reference : https://github.com/hyperledger/aries-rfcs/tree/master/features/0023-did-exchange#2-exchange-response
```
Comment (rolsonquadras) [inline test/bdd/features/didexchange_e2e.feature]: Do we need suffix "#2-exchange-response" in the url ?

File: test/bdd/features/didexchange_e2e.feature [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+@all
+Feature: Decentralized Identifier(DID) exchange between the agents
+
+  @didexchange_e2e
+  Scenario: did exchange e2e flow
+    Given "Alice" agent is running on "localhost" port "random"
+    Given "Bob" agent is running on "localhost" port "random"
+    And   "Alice" create invitation
```
Comment (sandrask) [inline test/bdd/features/didexchange_e2e.feature]: "Alice" crates invitation

File: test/bdd/features/didexchange_e2e.feature [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    And   "Bob" receive invitation from "Alice"
```
Comment (sandrask) [inline test/bdd/features/didexchange_e2e.feature]: "Bob" receives invitation

File: test/bdd/features/didexchange_e2e.feature (hunk 6 above)
Comment (fqutishat) [reply test/bdd/features/didexchange_e2e.feature]: done

**Pull request merged**